### ISSUE TITLE
mxnet very bad accuracy compared to other packages (simply when learning a*b*c!)

### ISSUE THREAD

**Issue by edebonneuil** [apache/incubator-mxnet#4798]
Hello,

I find that learning how to do an addition works well in mxnet but not how to do a multiplication. Can anyone see what training parameters to use? I tried various learning rates, batch sizes etc with overall very bad performance compared to other deeplearning packages...

Here is a little example in R:
library(mxnet)
x=matrix(runif(3000),1000,3); y=x[,1]*x[,2]*x[,3]; #training set of 1000 samples
x2=matrix(runif(3000),1000,3); y2=x2[,1]*x2[,2]*x2[,3]; #testing set
m = mx.mlp(x,y,hidden_node=100, out_node=1,out_activation="rmse", learning.rate=0.01); #training
print(sqrt(mean(predict(m, x2)-y2)^2)) #test: around 0.05, i.e. **about 40% error!** 

example of comparison: H2O:
h2o.init()
m=h2o.deeplearning(x=1:3,y=4,training_frame=as.h2o(cbind(x,y)))
print(sqrt(mean( (as.data.frame(predict(object=m,y=4,newdata=as.h2o(x2)))$predict-y2)^2 )))
#relative error on the order of 10%. #of note, even that is much; some other packages get 1% error.

**Comment by mayer79** [comment]
Your learning task is non-trivial due to the extreme skew of y (many values around 0, few large), so tit seems that using default optimizer resulted in some local optimum far away from the true optimium. Furthermore, 100 hidden nodes are quite a lot (low signal to noise ratio). 

Your example in R coding style could look as follows
```
library(mxnet)

# training set of 1000 samples
set.seed(8756)
x <- matrix(runif(3000), 1000, 3)
y <- x[, 1] * x[, 2] * x[, 3]

# validation set 
x2 <- matrix(runif(3000), 1000, 3)
y2 <- x2[, 1] * x2[, 2] * x2[, 3] 

# model
mx.set.seed(1)
fit <- mx.mlp(data = x, 
              label = y,
              hidden_node = 100,
              out_node = 1, 
              out_activation = "rmse", 
              learning.rate = 0.01,
              eval.metric = mx.metric.rmse,
              num.round = 350,
              array.batch.size = 30,
              optimizer = "adam")
sqrt(mean(predict(fit, x2) - y2)^2) # 0.0018

plot(y2 ~ t(predict(fit, x2)))
abline(0, 1)
```

**Comment by phunterlau** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
